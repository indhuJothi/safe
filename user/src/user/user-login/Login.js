import React from "react";
import "./LogReg.css";
import validateLogin from "../../service/service";
import { getPassword } from "../../service/service";
import { getMobile } from "../../service/service";
import { getUsername } from "../../service/service";
import { getUseremail } from "../../service/service"
import { Redirect, withRouter } from "react-router-dom";
import { userContext } from "../../context/Context";
import Header from "../../common/header/Header";
import '../../common/header/Header.css'
import baseURL from '../../service/api'
import Swal from "sweetalert2";




class Login extends React.Component {
  static contextType = userContext;
  constructor(props) {
    super(props);
    this.state = {
      mobile: "",
      password: "",
      boolean: false,
      search: false,
      register: false,
      mobileErr: "",
      passErr: "",
      userName: "",
      userEmail: "",
      gotoRegister:""
    };
    this.handleChange = this.handleChange.bind(this);
    this.submit = this.submit.bind(this);
    this.handleSignup = this.handleSignup.bind(this);
  }
  handleSignup(e) {
    e.preventDefault();
    this.setState({
      register: true,
    });
  }
  handleChange(event) {
    event.preventDefault();
    const mobile = event.target.name;
    const password = event.target.password;
    this.setState({
      [mobile]: event.target.value,
      [password]: event.target.value,
    });
  }

  submit(e) {
    e.preventDefault();
    let result, password, mobile, name, email;
    let mobileResult,passwordResult
    let mobileRegex = /^[6-9]\d{9}$/;
    let passRegex = /^[A-Za-z0-9@\s]{3,15}$/;
 
    if (passRegex.test(this.state.password)) {
      this.setState({
        passErr: "",
        gotoRegister:""
      });
      passwordResult=false
    }
    else{
      passwordResult=true
      this.setState({
        passErr:"Please Enter a valid password",
        gotoRegister:""
      })
    }
    if (mobileRegex.test(this.state.mobile)) {
      mobileResult = false
      this.setState({
        mobileErr: "",
        gotoRegister:""
      });
    }
    else{
      mobileResult=true
      this.setState({
        mobileErr:"Please Enter a valid mobile number",
        gotoRegister:""
      })
    }
    let newUserDetails = {
      mobile:parseInt(this.state.mobile),
      password: this.state.password,
    };
     let data
    if((mobileResult ===false) && (passwordResult===false))
    {
     this.loginUser(newUserDetails)
      .then((response) => data=response.data)
      .then((data) => {
        let token  = data.token;
        let user = data.user
       
        if(data !=="error" && data!=="Not found")
        {
          console.log(data.user)
        sessionStorage.setItem("authToken", token);
        sessionStorage.setItem("user",user)
         this.setState({
           search:true,
           mobileErr:"",
           passErr:""
         })
         this.props.history.push('/search')
        }
        else if(data ==="Not found")
        {
          this.setState({
            gotoRegister:"You don't have an account please register and then login",
            search:false
          })
          
        }
        else {
          this.setState({
            mobileErr:"",
            passErr:"Please enter a valid password",
            gotoRegister:""
          })
        }
     
        });
    }
  }
  loginUser(newUserDetails) {
    let apiUrl = "/users/login";
    
    return baseURL.post(apiUrl, newUserDetails, {
      headers: {
        "Content-Type": "application/json"
      },
    });
  }
  
  render() {
  
    let user = this.props.isuserpass;
 
    return (
      <div>
        <Header />
        {sessionStorage.getItem("authToken") ?
         (
          <Redirect to="/search"></Redirect>
         ) : (
          <form
            onSubmit={(e) => {
              this.submit(e);
            }}
          >
            <div className="base-container">
              <div class="MainContainer center">
                <button
                  class="button"
                >
                  Login
                </button>
                <button
                  onClick={(e) => {
                    this.handleSignup(e);
                  }}
                  class="button"
                >
                  Signup
                </button>
                <div className="formheader">Login</div>
                <p class="error">{this.state.gotoRegister}</p>
                <div className="form">
                  <div>
                    <label htmlFor="Mobile">Mobile</label>
                    <input
                      type="text"
                      name="mobile"
                      placeholder="MobileNo"
                      onChange={(event) => {
                        this.handleChange(event);
                      }}
                    />
                    <div class="error">{this.state.mobileErr}</div>
                  </div>
                  <div>
                    <label htmlFor="password">Password</label>
                    <input
                      type="password"
                      name="password"
                      placeholder="password"
                      onChange={(event) => {
                        this.handleChange(event);
                      }}
                    />
                    <div class="error">{this.state.passErr}</div>
                  </div>
                </div>
                <div>
                  <input type="submit" class="submitbtn"></input>
                </div>
              </div>
            </div>
          </form>
        )}
        {this.state.register && <Redirect to="/register"></Redirect> }
        {/* {this.state.search && <Redirect to="/search" />} */}
        {/* {this.state.search && user(this.state.password,this.state.userName, this.state.userEmail, this.state.mobile)} */}
      </div>
    );
  }
}

export default withRouter(Login);
