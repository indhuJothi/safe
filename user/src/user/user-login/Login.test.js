import { render, screen } from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom'
import { fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Login from './Login';

test('renders learn react link', () => {
  render(
  <MemoryRouter>
  <Login/>
  </MemoryRouter>
  );
  const loginElement = screen.getByTestId('test-login');
  const mobile = screen.getByPlaceholderText('MobileNo');
  const password = screen.getByPlaceholderText('password');
  expect(loginElement).toBeInTheDocument();
  expect(mobile).toBeInTheDocument();
  expect(mobile).toHaveAttribute('type',"text");
  expect(mobile).toHaveValue('')
  expect(password).toBeInTheDocument();
  expect(password).toHaveAttribute('type',"password");
  expect(password).toHaveValue('')
});


test('renders learn react link', () => {
    render(
    <MemoryRouter>
    <Login />
    </MemoryRouter>
    );
    const mobileVal = screen.getByPlaceholderText(/MobileNo/i); 
    const passwordVal = screen.getByPlaceholderText(/password/i);
    expect(mobileVal.value).toBe("")
    fireEvent.change(mobileVal,{target:{value:'7339467878'}})
    expect(mobileVal.value).toBe('7339467878')
    expect(passwordVal.value).toBe('')
    fireEvent.change(passwordVal,{target:{value:"password@123"}})
    expect(passwordVal.value).toBe('password@123')
})
   


test('get snapshot', () => {
    render(
    <MemoryRouter>
    <Login/>
    </MemoryRouter>
    );
  const val ='one'
  const tree = renderer.create(<MemoryRouter><Login/></MemoryRouter>).toJSON();
  expect(tree).toMatchSnapshot()
  console.log(tree);
})
