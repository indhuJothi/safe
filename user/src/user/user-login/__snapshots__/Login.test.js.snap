// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get snapshot 1`] = `
<div
  data-testid="test-login"
>
  <div>
    <div
      class="body"
    >
      <div
        class="header"
      >
        <span
          class="apptitle"
        >
          Bus Booking App
        </span>
      </div>
    </div>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="base-container"
    >
      <div
        class="MainContainer center"
      >
        <button
          class="button"
        >
          Login
        </button>
        <button
          class="button"
          onClick={[Function]}
        >
          Signup
        </button>
        <div
          className="formheader"
        >
          Login
        </div>
        <p
          class="error"
        >
          
        </p>
        <div
          className="form"
        >
          <div>
            <label
              htmlFor="Mobile"
            >
              Mobile
            </label>
            <input
              name="mobile"
              onChange={[Function]}
              placeholder="MobileNo"
              type="text"
              value=""
            />
            <div
              class="error"
            >
              
            </div>
          </div>
          <div>
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
            />
            <div
              class="error"
            >
              
            </div>
          </div>
        </div>
        <div>
          <input
            class="submitbtn"
            data-testid="test-submit"
            type="submit"
          />
        </div>
        <br />
        Don't have an account? 
        <a
          class="register"
          href="/register"
        >
          Register here
        </a>
      </div>
    </div>
  </form>
</div>
`;
